enhancements/helm3/console.md

```diff
@@ -0,0 +1,242 @@
+  disabled: true
+```
+
+An operator would watch for changes on them and reconfigure the chart repository proxy which is used to power the UI experience in OpenShift today.. 
+
+Please note, the console backend already implements a few helm endpoints (including the chart proxy). In future, we plan  extract them into a separate service to make the functionality more modular, thereby decoupling the API contract from the scenario-specific implementations. 
+
+Cluster admins would be able to easily manage `HelmChartRepositories` using the UI:
+
+![Helm Cluster Configuration](assets/openshift-administration-cluster-settings.png)
+
+or via CLI:
+
+```shell
+$ oc get helmchartrepositories
+NAME          AGE
+cluster       3h30m
+```
+
+In a case of multiple chart repositories, the console should be modified so that either allow:
+* editing all chart repository instances at once through aggregated YAML document
+* add/edit/removal of individual instances
+
+
+Adding an additional chart repositories via CLI follows the usual k8s pattern:
+
+```shell
+$ cat <<EOF | oc apply -f -
+apiVersion: helm.kubernetes.io/v1
+kind: HelmChartRepository
+metadata:
+  name: stable
+spec:
+  url: https://kubernetes-charts.storage.googleapis.com
+
+  displayName: Public Helm stable charts
+
+  description: Public Helm stable charts hosted on HelmHub
+---
+apiVersion: helm.kubernetes.io/v1
+kind: HelmChartRepository
+metadata:
+  name: incubator
+spec:
+  url: https://kubernetes-charts-incubator.storage.googleapis.com
+
+  displayName: Public Helm charts in incubator state
+EOF 
+
+$ kubectl get helmchartrepositories
+NAME          AGE
+cluster       3h30m
+stable        1m
+incubator     1m
+```
+
+Chart repository proxy will use all configured chart repositories and deliver to the UI an aggregated index file. If needed by some future usecase, UI would be able read helm chart repository configuration and perhaps even talk to the individual chart repositories directly.
+
+#### Taking the HelmChartRepository API upstream
+
+##### Why
+
+Enable admins to configure `Helm` chart repositories in the cluster which would be available for any user accessing the cluster, without neccessarily requiring the user to do a `helm repo add`.
+
+
+#### How
+
+1) We plan to use the [ChartRepository Config](https://github.com/helm/helm/blob/master/pkg/repo/chartrepo.go#L42) data structure in the proposed CRD, which is also used by the Helm CLI to list/add chart repositories.
+
+2) Given that the `Helm` CLI manages chart repository configurations in the local filesystem only, we shall propose upstream changes in the [Helm CLI Repository](https://github.com/helm/helm) to optionally make the CLI aware of `HelmChartRepository` CRs in the current cluster context.
+
+As a consequence, `helm repo list` would not only list local chart repositories from the local file system, it would also list chart repositories represented as Kubernetes objects in the cluster.
```

Comment:
this sounds good, but I would expect that future API to exist in the helm API namespace and not the Kubernetes.io namespace.